latch_cli/utils.py

```diff
@@ -115,27 +116,44 @@ def with_si_suffix(num, suffix="B", styled=False):
 def hash_directory(dir_path: Path) -> str:
     m = hashlib.new("sha256")
     m.update(current_workspace().encode("utf-8"))
-    for containing_path, dirnames, fnames in os.walk(dir_path):
+
+    ignore_file = dir_path / ".dockerignore"
+    exclude: List[str] = []
+    try:
+        for l in ignore_file.open("r"):
+            l = l.strip()
+
+            if l == "":
+                continue
+            if l[0] == "#":
+                continue
+
+            exclude.append(l)
+    except FileNotFoundError:
+        print("No .dockerignore file found --- including all files")
+
+    paths = list(exclude_paths(dir_path, exclude))
+    paths.sort()
+
+    for item in paths:
         # for repeatability guarantees
-        dirnames.sort()
-        fnames.sort()
-        for filename in fnames:
-            path = Path(containing_path).joinpath(filename)
-            m.update(str(path).encode("utf-8"))
-            file_size = os.path.getsize(path)
-            if file_size < FILE_MAX_SIZE:
-                with open(path, "rb") as f:
-                    m.update(f.read())
-            else:
-                print(
-                    "\x1b[38;5;226m"
-                    f"WARNING: {path.relative_to(dir_path.resolve())} is too large "
-                    f"({with_si_suffix(file_size)}) to checksum, skipping."
-                    "\x1b[0m"
-                )
-        for dirname in dirnames:
-            path = Path(containing_path).joinpath(dirname)
-            m.update(str(path).encode("utf-8"))
+        p = Path(dir_path / item)
+        if p.is_dir():
+            m.update(str(p).encode("utf-8"))
+            continue
+
+        m.update(str(p).encode("utf-8"))
+        file_size = os.path.getsize(p)
```

Comment:
```suggestion
        file_size = p.stat().st_size
```
same thing but from pathlib
see https://github.com/python/cpython/blob/3eb12df8b526aa5a2ca6b43f21a1c5e7d38ee634/Lib/genericpath.py#L60